catalog/to-consul/syncer.go

```diff
@@ -176,10 +176,10 @@ func (s *ConsulSyncer) watchReapableServices(ctx context.Context) {
 		}, backoff.WithContext(backoff.NewExponentialBackOff(), ctx))
 		if err != nil {
 			s.Log.Warn("error querying services, will retry", "err", err)
-			continue
```

Comment:
Previously, this `continue` would have us immediately restarting the `for` loop. Instead we should fall down into the `select` statement where we retry after a defined period of time